src/monattack.cpp

```diff
@@ -2730,72 +2730,77 @@ void mattack::taze( monster *z, Creature *target )
 
 bool mattack::smg(monster *z)
 {
-    const std::string ammo_type("9mm");
-    // Make sure our ammo isn't weird.
-    if (z->ammo[ammo_type] > 1000) {
-        debugmsg("Generated too much ammo (%d) for %s in mattack::smg", z->ammo[ammo_type], z->name().c_str());
-        z->ammo[ammo_type] = 1000;
-    }
+    item gun( "hk_mp5", calendar::turn );
+    item mag( "mp5mag", calendar::turn ); // @todo refactor use item::magazine_default() from #15217
+
+    itype_id atype = default_ammo( mag.ammo_type() );
+
+    item ammo( atype, calendar::turn ); // fetch matching ammo from the turret's supply
+    ammo.charges = std::max( std::min( z->ammo[ atype ], int( mag.ammo_capacity() ) ), 0 );
+
+    mag.contents.push_back( std::move( ammo ) );
+    gun.contents.push_back( std::move( mag ) );
+
+    npc fake = make_fake_npc( z, 16, 8, 8, 12 );
+    fake.skillLevel( gun.gun_skill() ).level( 8 );
+    fake.skillLevel( skill_gun ).level( 4 );
 
     Creature *target = nullptr;
 
     if (z->friendly != 0) {
-        // Attacking monsters, not the player!
-        int boo_hoo;
-        target = z->auto_find_hostile_target( 18, boo_hoo );
-        if( target == nullptr ) {// Couldn't find any targets!
-            if(boo_hoo > 0 && g->u.sees( *z ) ) { // because that stupid oaf was in the way!
-                add_msg(m_warning, ngettext("Pointed in your direction, the %s emits an IFF warning beep.",
-                                            "Pointed in your direction, the %s emits %d annoyed sounding beeps.",
-                                            boo_hoo),
-                        z->name().c_str(), boo_hoo);
+        int avoid; // number of friendly targets that were avoided
+        target = z->auto_find_hostile_target( gun.gun_range(), avoid );
```

Comment:
Whether or not we go with this PR it might be interesting to build upon this idea now that `player::fire_gun` supports arbitrary burst sizes? Turrets could snipe at longer range and engage full-auto at shorter distances?